go/sqltypes/type.go

```diff
@@ -245,6 +251,22 @@ func MySQLToType(mysqlType, flags int64) (typ querypb.Type, err error) {
 	return modifyType(result, flags), nil
 }
 
+//TypeEquivalenceCheck returns whether two types are equivalent.
```

Comment:
I suggest renaming this to AreTypesEquivalent, the general practice for functions that return booleans is to name the function like a question.